src/main/java/org/sagebionetworks/web/client/widget/discussion/DiscussionThreadListWidget.java

```diff
@@ -59,12 +66,31 @@ public void configure(String forumId, Boolean isCurrentUserModerator, Set<Long>
 		this.moderatorIds = moderatorIds;
 		offset = 0L;
 		this.forumId = forumId;
+		invokeCheckForInViewAndLoadData = new Callback() {
+			@Override
+			public void invoke() {
+				checkForInViewAndLoadData();
+			}
+		};
 		loadMore();
 		DiscussionThreadCountAlert threadCountAlert = ginInjector.getDiscussionThreadCountAlert();
 		view.setThreadCountAlert(threadCountAlert.asWidget());
 		threadCountAlert.configure(forumId);
 	}
 
+	public void checkForInViewAndLoadData() {
+		if (!view.isLoadMoreAttached()) {
```

Comment:
Shouldn't this loop also end if the load more is no longer visible?  (!view.getLoadMoreVisibility())